### PR TITLE
Error message for allow_in_graph decorator and arbitrary function combo

### LINKED ISSUE #103615: Can't call allow_in_graph inside of a function being torch.compile'd
### 🐛 Describe the bug

We are unable to call allow_in_graph inside of a function being torch.compile'd with fullgraph=True and the error message is confusing. We should either allow this or improve the error message.
```py
import torch
import torch._dynamo

@torch.compile(backend='aot_eager', fullgraph=True)
def f(x):
    @torch._dynamo.allow_in_graph
    def apply_func(x):
        return x.clone()
    
    return apply_func(x)

x = torch.randn([], requires_grad=True)

f(x)
```
produces: 
> Unsupported: call_function allow_in_graph in skip_files /raid/rzou/pt/debug-cpu4/torch/_dynamo/__init__.py

Discovered with @drisspg 

### Error logs

_No response_

### Minified repro

_No response_

### Versions

main

cc @ezyang @msaroufim @bdhirsh @anijain2305 @chauhang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @wconstab

### PR DESCRIPTION
**Pull request by DuyguA**
Fixes #103615

Quick error message for non-allowed allow_in_graph decorator and arbitrary function combo.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

**Comment by DuyguA** [comment]
cc @mlazos  and @anijain2305

File: torch/_dynamo/variables/functions.py [hunk 1]
```diff
@@ -689,9 +689,14 @@ def wraps(fn):
             return variables.LambdaVariable(wraps)
         else:
             try:
+                print("here")
```
Comment (mlazos) [inline torch/_dynamo/variables/functions.py:692]: remove debug prints

Comment (DuyguA) [reply torch/_dynamo/variables/functions.py]: Forgot about them, thanks for point out @mlazos !

Comment (DuyguA) [reply torch/_dynamo/variables/functions.py]: Gentle ping @mlazos  , made the requested changes :blush:

File: torch/_dynamo/variables/functions.py [hunk 2]
```diff
@@ -725,6 +725,13 @@ def wraps(fn):
                     )
                     # also warn on it because most users won't see the graph break message
                     torch._dynamo.utils.warn_once(msg)
+            if self.value.__qualname__ == "allow_in_graph":
+                msg = (
+                    "Graph break due to usage of an arbitrary function with "
+                    "allow_in_graph decorator. Usage of an arbitrary function "
+                    "allowed in the graph doesn't guarantee that the code being "
+                    "traced is represented by the full FX graph."
+                )
```
[the remark this answers is not in the document]
Comment (DuyguA) [reply torch/_dynamo/variables/functions.py:733]: Thanks for the suggestion, I changed the message. Hope ready for merge now 😊

[the remark this answers is not in the document]
Comment (DuyguA) [reply torch/_dynamo/variables/functions.py:733]: Thanks for the suggestion @anijain2305  , I changed the message. Hope ready for merge now 😊

**Comment by DuyguA** [comment]
Gentle ping @mlazos  and @anijain2305

**Review by anijain2305 — APPROVED** [review]

**Pull request closed without merging**